app/account/components/AccountDetails/components/AvatarUpdate/AvatarUpdate.tsx

```diff
@@ -0,0 +1,173 @@
+"use client"
+
+import { updateUserInfo } from "@api"
+import { Button, ModalOverlay } from "@components"
+import { User } from "@interfaces"
+import { useUserStore } from "@stores"
+import { customFetch } from "@utils"
+import React, { useCallback, useState } from "react"
+import Cropper from "react-easy-crop"
+import "./avatarUpdate.scss"
+import { getCroppedImg } from "./cropUtils"
+
+const AvatarUpdate = () => {
+  const fetchUserData = useUserStore((state) => state.fetchUserData)
+  const [imageSrc, setImageSrc] = useState<string | null>(null)
+  const [crop, setCrop] = useState({ x: 0, y: 0 })
+  const [zoom, setZoom] = useState(1)
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState<null | { width: number; height: number; x: number; y: number }>(null)
+  const [isCropping, setIsCropping] = useState(false)
+  const [userMsg, setUserMsg] = useState<string>()
+  const fileInputRef = React.useRef<HTMLInputElement | null>(null)
+  const user = useUserStore((state) => state.user)
+  const { username } = (user as User) ?? {}
+  const msgShowDuration = 2000
+
+  const closeModal = () => {
+    // reset cropper
+    setImageSrc(null)
+    setCrop({ x: 0, y: 0 })
+    setZoom(1)
+    setCroppedAreaPixels(null)
+    setUserMsg(undefined)
+
+    if (fileInputRef.current) {
+      fileInputRef.current.value = ""
+    }
+  }
+
+  const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0]
+
+    if (file) {
+      const reader = new FileReader()
+
+      reader.onload = () => setImageSrc(reader.result as string)
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const handleCropComplete = useCallback((croppedArea: any, croppedAreaPixels: any) => {
+    setCroppedAreaPixels(croppedAreaPixels)
+  }, [])
+
+  const showErrorMsg = () => {
+    setUserMsg("Failed to update avatar. Please try again.")
+
+    setTimeout(() => {
+      setUserMsg(undefined)
+    }, msgShowDuration)
+  }
+
+  const handleCloudinaryUploadSuccess = async (url: string) => {
+    try {
+      setUserMsg("Avatar updated successfully in the cloud! Updating Nexus dawn...")
```

Comment:
I like the use of the `ModalOverlay` for the `userMsg`!

I'm curious if you think we should trim down to one message? I notice the first one flashes for maybe 1/2 a second, not enough to read completely, and the second stays visible for the duration you've desiganted, which is perfect. Let me know what you think!

Reply:
My thought process was that since the saving process is multi-stage it made sense to also have messages for each stage, kind of like when resources are loading when starting discord. I don't really see a problem with it being shown for a very short time, but let me know if you'd like me to update

Reply:
Nope, works fine as is. Curious what you thought about it, but I could go either way with this one. 👍